src/mongo_odm/nvp.hpp

```diff
@@ -74,17 +159,593 @@ struct Nvp {
+        const std::int64_t& bitmask) const {
+        return {*static_cast<const NvpT*>(this), bitmask, "$bitsAnyClear"};
+    }
+
+    /**
+     * Creates a query that uses the $bitsAnyClear operator to check a series of bits, given as bit
+     * positions. This function has two positional arguments to distinguish from the signature that
+     * takes a bit mask (see above)
+     * @param pos1  The first bit position to check
+     * @param pos2  The second bit position to check
+     * @param positions...  Variadic argument containing further bit positions.
+     * @returns A ComparisonExpr representing this query
+     */
+    template <typename U = no_opt_type,
+              typename = typename std::enable_if<std::is_integral<U>::value>::type,
+              typename... Args>
+    constexpr ComparisonValueExpr<NvpT, std::int64_t> bits_any_clear(std::int64_t pos1,
+                                                                     std::int64_t pos2,
+                                                                     Args... positions) const {
+        return {*static_cast<const NvpT*>(this), bit_positions_to_mask(pos1, pos2, positions...),
+                "$bitsAnyClear"};
+    }
+
+    constexpr UpdateExpr<NvpT, no_opt_type> set_on_insert(const no_opt_type& val) const {
+        return {*static_cast<const NvpT*>(this), val, "$setOnInsert"};
+    }
+
+    /* Update operators */
+
+    /**
+     * Creates an expression that unsets the current field. The field must be of optional type.
+     * @returns An UnsetExpr that unsets the current field.
+     */
+    template <typename U = T, typename = typename std::enable_if<is_optional<U>::value>::type>
+    constexpr UnsetExpr<NvpT> unset() const {
+        return {*static_cast<const NvpT*>(this)};
+    }
+
+    /**
+     * Creates an expression that uses the $min operator to only update a field if the new value is
+     * lower than the current value.
+     * @param val   The (tentative) new value.
+     * @returns     An UpdateExpression with the $min operator.
+     */
+    constexpr UpdateExpr<NvpT, no_opt_type> min(const no_opt_type& val) const {
+        return {*static_cast<const NvpT*>(this), val, "$min"};
+    }
+
+    /**
+     * Creates an expression that uses the $max operator to only update a field if the new value is
+     * greater than the current value.
+     * @param val   The (tentative) new value.
+     * @returns     An UpdateExpression with the $max operator.
+     */
+    constexpr UpdateExpr<NvpT, no_opt_type> max(const no_opt_type& val) const {
+        return {*static_cast<const NvpT*>(this), val, "$max"};
+    }
+
+    /**
+     * Creats an expression that sets a date value to the current date.
+     * This is only enabled for std::chrono::time/duration values, as well as b_date.
+     */
+    template <typename U = no_opt_type>
+    constexpr typename std::enable_if<is_date<U>::value, CurrentDateExpr<NvpT>>::type current_date()
```

Comment:
Instead of making this a member function, can this be an overload of `operator=()` for type `odm_current_date_t` and then also provide a `const odm_current_date_t odm_current_date;`